test/cql-pytest/test_select_from_mutation_fragments.py

```diff
@@ -182,7 +182,11 @@ def test_many_partition_scan(cql, test_keyspace, scylla_only):
         requests.post(f'{nodetool.rest_api_url(cql)}/system/drop_sstable_caches')
 
         # the real test here is that this scan completes without problems
-        res_all = list(cql.execute(f"SELECT pk, mutation_source, mutation_fragment_kind, metadata FROM MUTATION_FRAGMENTS({test_table});"))
+        # since this scan is very slow, we create an extra patient cql connection,
+        # with an abundant 10 minutes timeout
+        ep = cql.hosts[0].endpoint
+        with util.cql_session(ep.address, ep.port, False, 'cassandra', 'cassandra', 600) as patient_cql:
```

Comment:
The scan in `SELECT FROM MUTATION_FRAGMENTS()` is an especially slow one: it obtains a list of partitions from the table, then produces the content for each of them individually. It is sort of like a scan via secondary index. For N partitions, this is N/1000 range scans + N single partition reads.

Reply:
Right :-( Maybe this test has a lot of data? What's special about this test that only it needs this increase - maybe we need to increase the 120 seconds to 600 for all tests? :-(
It was already ridiculous that 10 seconds (the default) was not enough and I had to increase it to 120 seconds which "should have been enough for anyone" (famous last words)...

Reply:
The code has an internal limit of 1000 partitions per posting list. To exercise the logic which switches between lists, the test writes 1300 (tiny) partitions. This is not a lot, but like I mentioned above, the algorithm is veeerrrryyyy slow. I spent no time on trying to make it fast, I focused on making it simple and correct, because this is a rarely used query on a rarely used statement.

Reply:
Note that on release/dev, the scan is completely reasonable, the entire test finishes in a few seconds. The problem is on debug, especially ARM debug.